checks.d/docker_daemon.py

```diff
@@ -540,14 +529,43 @@ def _submit_healthcheck_sc(self, container):
         tags = self._get_tags(container, CONTAINER)
         self.service_check(HEALTHCHECK_SERVICE_CHECK_NAME, status, tags=tags)
 
+    def _report_container_count(self, containers_by_id):
+        """Report container count per state"""
+        m_func = FUNC_MAP[GAUGE][self.use_histogram]
+
+        per_state_count = defaultdict(int)
+
+        filterlambda = lambda ctr: not self._is_container_excluded(ctr)
+        containers = list(filter(filterlambda, containers_by_id.values()))
+
+        for ctr in containers:
+            per_state_count[ctr.get('State', '')] += 1
+
+        for state in per_state_count:
+            if state:
+                m_func(self, 'docker.container.count', per_state_count[state], tags=['container_state:%s' % state.lower()])
+
+    def _report_volume_count(self):
+        """Report volume count per state (dangling or not)"""
+        m_func = FUNC_MAP[GAUGE][self.use_histogram]
+
+        attached_volumes = self.docker_client.volumes(filters={'dangling': False})
+        dangling_volumes = self.docker_client.volumes(filters={'dangling': True})
+        attached_count = len(attached_volumes['Volumes'])
+        dangling_count = len(dangling_volumes['Volumes'])
+        m_func(self, 'docker.volume.count', attached_count, tags=['volume_state:attached'])
+        m_func(self, 'docker.volume.count', dangling_count, tags=['volume_state:dangling'])
+
     def _report_image_size(self, images):
         for image in images:
-            tags = self._get_tags(image, IMAGE)
-            if 'VirtualSize' in image:
-                self.gauge('docker.image.virtual_size', image['VirtualSize'], tags=tags)
-            if 'Size' in image:
-                self.gauge('docker.image.size', image['Size'], tags=tags)
-
+            try:
+                tags = self._get_tags(image, IMAGE)
+                if 'VirtualSize' in image:
+                    self.gauge('docker.image.virtual_size', image['VirtualSize'], tags=tags)
+                if 'Size' in image:
+                    self.gauge('docker.image.size', image['Size'], tags=tags)
+            except:
+                import ipdb;ipdb.set_trace()
```

Comment:
This should be on 2 line (that why the CI flake8 tests are falling). End I'm not sure we want to keep it, looks a lot like some debug lines.